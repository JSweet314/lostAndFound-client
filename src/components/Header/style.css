.header {
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  background-color: #000;
  color: #fff;
  width: 100%;
  height: 10rem;
}

.header__h1 {
  color: #fff;
  display: inline-block;
  margin: 0;
  font-size: 5rem;
  font-weight: 100;
  text-align: center;
  text-decoration: none;
}

.header__nav-list-item {
  display: inline-block;
  margin: 1rem;
}

.header__username,
.header__nav-link {
  font-size: 2rem;
  color: #fff;
  text-decoration: none;
}

.header__username {
  color: #C47E59;
}

.header__h1:hover,
.header__nav-link:hover {
  color: #C47E59;
}