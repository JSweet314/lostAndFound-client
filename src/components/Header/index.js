import React from 'react';
import {Link} from 'react-router-dom';
import PropTypes from 'prop-types';
import './style.css';

const Header = ({ loggedIn, username }) => {
  
  return (
    <header className='header'>
      <h1 className='header__h1'>Lost & Found</h1>
      <ul className='header__nav-list'>
        <li className='header__nav-list-item'>
          <Link className='header__nav-link' to='/'>Home</Link>
        </li>
        <li className='header__nav-list-item'>
          {loggedIn ? 
            <Link className='header__nav-link' to='/'>Log Out</Link> :
            <Link className='header__nav-link' to='/forms/login'>Log In</Link>
          }
        </li>
        <li className='header__nav-list-item'>
          {loggedIn ? <div className='header__nav-link'>{username}</div> : (
            <Link className='header__nav-link' to='/forms/signup'>Sign Up</Link>
          )}
        </li>
      </ul>
    </header>
  );
};

Header.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  username: PropTypes.string
};

export default Header;