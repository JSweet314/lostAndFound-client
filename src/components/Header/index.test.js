import React from 'react';
import {shallow} from 'enzyme';
import Header from './index';

describe('Header', () => {
  let wrapper = shallow(<Header/>);
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});