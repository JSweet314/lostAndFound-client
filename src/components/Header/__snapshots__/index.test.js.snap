// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    handleLogOut={[Function]}
    loggedIn={false}
    username=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="header__h1"
          replace={false}
          to="/"
>
          Lost & Found
</Link>,
        <ul
          className="header__nav-list"
>
          <li
                    className="header__nav-list-item"
          >
                    <Link
                              className="header__nav-link"
                              replace={false}
                              to="/user-forms/login"
                    >
                              Log In
                    </Link>
          </li>
          <li
                    className="header__nav-list-item"
          >
                    <Link
                              className="header__nav-link"
                              replace={false}
                              to="/user-forms/signup"
                    >
                              Sign Up
                    </Link>
          </li>
</ul>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Lost & Found",
          "className": "header__h1",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Lost & Found",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <li
              className="header__nav-list-item"
>
              <Link
                            className="header__nav-link"
                            replace={false}
                            to="/user-forms/login"
              >
                            Log In
              </Link>
</li>,
            <li
              className="header__nav-list-item"
>
              <Link
                            className="header__nav-link"
                            replace={false}
                            to="/user-forms/signup"
              >
                            Sign Up
              </Link>
</li>,
          ],
          "className": "header__nav-list",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="header__nav-link"
                replace={false}
                to="/user-forms/login"
>
                Log In
</Link>,
              "className": "header__nav-list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Log In",
                "className": "header__nav-link",
                "replace": false,
                "to": "/user-forms/login",
              },
              "ref": null,
              "rendered": "Log In",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="header__nav-link"
                replace={false}
                to="/user-forms/signup"
>
                Sign Up
</Link>,
              "className": "header__nav-list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign Up",
                "className": "header__nav-link",
                "replace": false,
                "to": "/user-forms/signup",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="header__h1"
            replace={false}
            to="/"
>
            Lost & Found
</Link>,
          <ul
            className="header__nav-list"
>
            <li
                        className="header__nav-list-item"
            >
                        <Link
                                    className="header__nav-link"
                                    replace={false}
                                    to="/user-forms/login"
                        >
                                    Log In
                        </Link>
            </li>
            <li
                        className="header__nav-list-item"
            >
                        <Link
                                    className="header__nav-link"
                                    replace={false}
                                    to="/user-forms/signup"
                        >
                                    Sign Up
                        </Link>
            </li>
</ul>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Lost & Found",
            "className": "header__h1",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Lost & Found",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <li
                className="header__nav-list-item"
>
                <Link
                                className="header__nav-link"
                                replace={false}
                                to="/user-forms/login"
                >
                                Log In
                </Link>
</li>,
              <li
                className="header__nav-list-item"
>
                <Link
                                className="header__nav-link"
                                replace={false}
                                to="/user-forms/signup"
                >
                                Sign Up
                </Link>
</li>,
            ],
            "className": "header__nav-list",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="header__nav-link"
                  replace={false}
                  to="/user-forms/login"
>
                  Log In
</Link>,
                "className": "header__nav-list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Log In",
                  "className": "header__nav-link",
                  "replace": false,
                  "to": "/user-forms/login",
                },
                "ref": null,
                "rendered": "Log In",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="header__nav-link"
                  replace={false}
                  to="/user-forms/signup"
>
                  Sign Up
</Link>,
                "className": "header__nav-list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Up",
                  "className": "header__nav-link",
                  "replace": false,
                  "to": "/user-forms/signup",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header should match a snapshot when signed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    handleLogOut={[Function]}
    loggedIn={true}
    username="jon"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="header__h1"
          replace={false}
          to="/"
>
          Lost & Found
</Link>,
        <ul
          className="header__nav-list"
>
          <Link
                    className="header__nav-link"
                    replace={false}
                    to="/items"
          >
                    My Items
          </Link>
          <li
                    className="header__nav-list-item"
          >
                    <Link
                              className="header__nav-link"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                    >
                              Log Out
                    </Link>
          </li>
          <li
                    className="header__nav-list-item"
          >
                    <span
                              className="header__username"
                    >
                              jon
                    </span>
          </li>
</ul>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Lost & Found",
          "className": "header__h1",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Lost & Found",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="header__nav-link"
              replace={false}
              to="/items"
>
              My Items
</Link>,
            <li
              className="header__nav-list-item"
>
              <Link
                            className="header__nav-link"
                            onClick={[Function]}
                            replace={false}
                            to="/"
              >
                            Log Out
              </Link>
</li>,
            <li
              className="header__nav-list-item"
>
              <span
                            className="header__username"
              >
                            jon
              </span>
</li>,
          ],
          "className": "header__nav-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "My Items",
              "className": "header__nav-link",
              "replace": false,
              "to": "/items",
            },
            "ref": null,
            "rendered": "My Items",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="header__nav-link"
                onClick={[Function]}
                replace={false}
                to="/"
>
                Log Out
</Link>,
              "className": "header__nav-list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Log Out",
                "className": "header__nav-link",
                "onClick": [Function],
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Log Out",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="header__username"
>
                jon
</span>,
              "className": "header__nav-list-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "jon",
                "className": "header__username",
              },
              "ref": null,
              "rendered": "jon",
              "type": "span",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="header__h1"
            replace={false}
            to="/"
>
            Lost & Found
</Link>,
          <ul
            className="header__nav-list"
>
            <Link
                        className="header__nav-link"
                        replace={false}
                        to="/items"
            >
                        My Items
            </Link>
            <li
                        className="header__nav-list-item"
            >
                        <Link
                                    className="header__nav-link"
                                    onClick={[Function]}
                                    replace={false}
                                    to="/"
                        >
                                    Log Out
                        </Link>
            </li>
            <li
                        className="header__nav-list-item"
            >
                        <span
                                    className="header__username"
                        >
                                    jon
                        </span>
            </li>
</ul>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Lost & Found",
            "className": "header__h1",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Lost & Found",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="header__nav-link"
                replace={false}
                to="/items"
>
                My Items
</Link>,
              <li
                className="header__nav-list-item"
>
                <Link
                                className="header__nav-link"
                                onClick={[Function]}
                                replace={false}
                                to="/"
                >
                                Log Out
                </Link>
</li>,
              <li
                className="header__nav-list-item"
>
                <span
                                className="header__username"
                >
                                jon
                </span>
</li>,
            ],
            "className": "header__nav-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "My Items",
                "className": "header__nav-link",
                "replace": false,
                "to": "/items",
              },
              "ref": null,
              "rendered": "My Items",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="header__nav-link"
                  onClick={[Function]}
                  replace={false}
                  to="/"
>
                  Log Out
</Link>,
                "className": "header__nav-list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Log Out",
                  "className": "header__nav-link",
                  "onClick": [Function],
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Log Out",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="header__username"
>
                  jon
</span>,
                "className": "header__nav-list-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "jon",
                  "className": "header__username",
                },
                "ref": null,
                "rendered": "jon",
                "type": "span",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
